### PR TITLE
fix(#14137): s/ '#!/bin/bash' / '#!/usr/bin/env bash'

### PR DESCRIPTION
**Pull request by yesudeep**
Fixes: https://github.com/bazelbuild/bazel/issues/14137

### CODE REVIEW

**Review by ted-xie — CHANGES_REQUESTED** [review]
This PR is too huge; there are 300+ files here and even loading the diff actually causes my browser to stutter. Please break this up into sub-PRs (maybe alphabetically, with ~10 files changed per PR) if this is still affecting you. Breaking it up into sub-PRs will also help the debug/rollback process if changing the shebangs cause any issues (which should not be the case, but you never know).

Also worth noting that public Google style guides prefer `#!/bin/bash` as the GNU bash shebang. See here: https://chromium.googlesource.com/chromiumos/docs/+/HEAD/styleguide/shell.md#standard-selection

I'm not saying we have to treat the Chromium shell style guide as the golden truth, but there is precedent to use `#!/bin/bash`. I can imagine that keeping shell versions in sync for developer environments is the main reason for this, at least internally. For external-facing projects such as Bazel, I can see some benefit to allowing for the more flexible `#!/usr/bin env bash` style.

**Comment by ted-xie** [comment]
Please also provide a less esoteric commit message. I would imagine most Bazel users are familiar with the sed-style regex syntax, but something straightforward like `Replace all shebangs with #!/usr/bin/env bash` would be a lot easier to read and understand at a glance.

**Comment by ted-xie** [comment]
Some more thoughts on this:

Is there a reason to change _every_ shebang in every shell script? I think only a subset of these scripts are directly executed in a way that the shebang even matters (such as `./a/b/c/d.sh`). Maybe there's something we can change in the job orchestrator that transcends the problem altogether.

**Comment by aiuto** [comment]
I suggest breaking the work along clear boundaries rather than by size or name.

- get bazel to dist-build
- get tests to run
- the long tail of other stuff.

**Comment by sgowroji** [comment]
Hello @yesudeep, Could you please respond to the above [comments](https://github.com/bazelbuild/bazel/pull/14138#issuecomment-1132192936) and resolve the conflicts. Thanks!

File: src/main/native/windows/build_windows_jni.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -eu
```
Comment (aiuto) [inline src/main/native/windows/build_windows_jni.sh:1]: I'm not sure this is right for scripts that only run on windows.

**Comment by aiuto** [comment]
From the conversation thread, it is pretty clear we should just close this PR and try again in a more disciplined way.

**Pull request closed without merging**